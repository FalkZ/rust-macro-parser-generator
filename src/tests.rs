macro_rules! tests {
    (@debug: $name:ident, $t:ident, $str:literal) => {
        #[test]
        fn $name() {
            let r = crate::m1n::renderer::render($str).expect("couldn't compile file");
            insta::assert_debug_snapshot!(&r.$t);
        }
    };
    (@display: $name:ident, $t:ident, $str:literal) => {
        #[test]
        fn $name() {
            let r = crate::m1n::renderer::render($str).expect("couldn't compile file");
            insta::assert_display_snapshot!(&r.$t);
        }
    };
    // `()` indicates that the macro takes no argument.
    ($($name:ident: $str:literal),*) => {
        mod tokens {
            $(
                tests!(@debug: $name, tokens, $str);
            )*
        }
        mod statements {
            $(
                tests!(@debug: $name, statements, $str);
            )*
        }
        mod typescript {
            $(
                tests!(@display: $name, typescript, $str);
            )*
        }
    };
}

tests!(
    enums: "./examples/m1n/Enum.m1n",
    class: "./examples/m1n/Class.m1n",
    file: "./examples/m1n/file.m1n",
    math: "./examples/m1n/std/math.m1n",
    matcher: "./examples/m1n/match.m1n",
    assignment: "./examples/m1n/assignment.m1n",
    operator: "./examples/m1n/operator.m1n"
);
