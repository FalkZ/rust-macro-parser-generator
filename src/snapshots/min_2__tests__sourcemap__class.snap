---
source: src/tests.rs
expression: "&r.sourcemap"
---
import util, { pipe } from '@std/util';
import math from '@std/math';

export class Class {
private readonly PI = pipe(a).op(math['+'], b).op(math['+'], 0.012).end();

public Pi = 123;


private fn(a, b) { return math['+'](a,b);}

private one(b) { return pipe(a).op(math['+'], b).op(math['+'], c).end();}

public zero() { return ((()=>{throw new Error("hello 2")})());}

private multiline() { return console.log(`test space`, `multi

    line`);}

private nesting() { return one(fn(math['+'](a,math['-'](b,c)), 123.2));}



};


AAAA,AACO,OACH,IAAW,IAAE,YAAb
AAAA,OACA,UAAA
AAAA
AAAA
AAAA,AAAY,iBAChB,KAAI,KAAC,MAAC,AAAC,WAAE,EAAC,IAAA,AACN,WAAC,AAAC,EAAE
AAAA
AAAA,AAAI,OAEJ,KAAI,AAAC
AAAA
AAAA
AAAA,AAAI,QAEjB,EAAG,CAAA,GAAG,YAAI,AAAE,AAAC,UAAF,EAAI
AAAA
AAAA,AAAE,QAEjB,GAAI,CAAA,YAAI,KAAC,MAAC,AAAC,WAAE,EAAC,IAAA,AACV,WAAE
AAAA
AAAA,AAAE,OAEJ,gBAAQ,AAAC
AAAA
AAAA,AAAuC,QAEpD,qBAAa,AAAC,QAAQ,IAAI,cAAa,AAAC
AAAA
AAAA
AAAA
AAAA,AAE7B,QAEX,mBAAW,AAAC,IAAI,GAAG,AAAC,AAAC,UAAF,EAAG,AAAE,AAAC,AAAC,UAAF,EAAI,KAAG,AAAC,IAAI
AAAA
AAAA
AAAA
AAAA
